actions/gradle/publish/README.md

```diff
@@ -11,3 +11,271 @@ Before you use this publish Action, you will need to configure your Github proje
+                    if (name.size != 2) {
+                        throw StopExecutionException("Found incorrect file name: " + path)
+                    }
+                    var cls = name[0]
+                    var ext = name[1]
+                    if (cls.startsWith("-")) {
+                        cls = cls.substring(1)
+                    }
+                    artifact (base_dir + "/" + path) {
+                        classifier = cls
+                        extension = ext
+                    }
+                }
+            }            
+            pom {
+                name.set("test-java-project")
+                description.set("Adams test java project")
+                url.set("https://github.com/AdamKorcz/test-java-project")
+                licenses {
+                    license {
+                        name.set("MIT License")
+                        url.set("http://www.opensource.org/licenses/mit-license.php")
+                    }
+                }
+                developers {
+                    developer {
+                        id.set("adamkrocz")
+                        name.set("Adam K")
+                        email.set("Adam@adalogics.com")
+                    }
+                }
+                scm {
+                    connection.set("scm:git:git://github.com/adamkorcz/test-java-project.git")
+                    developerConnection.set("scm:git:ssh://github.com:simpligility/test-java-project.git")
+                    url.set("http://github.com/adamkorcz/test-java-project/tree/main")
+                }
+            }
+        }
+    }
+    repositories {
+        maven {
+            credentials {
+                username = System.getenv("MAVEN_USERNAME")
+                password = System.getenv("MAVEN_PASSWORD")
+            }
+            name = "test-java-project"
+            url = uri("https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/")
+        }
+    }
+}
+
+signing {
+    useGpgCmd()
+    sign(publishing.publications["maven"])
+}
+```
+
+You don't need to configure anything inside that code snippet; Adding them to your `build.gradle.kts` file is enough.
+
+### Add the publish action to your release workflow
+
+Before using the Gradle publish action, you should have a workflow that invokes the Gradle builder. It will look something like this:
+
+```yaml
+name: Publish Gradle with action
+on:
+  - workflow_dispatch
+
+permissions: read-all
+
+jobs:
+  build:
+    permissions:
+      contents: read
+      id-token: write
+      actions: read
+      packages: read
+    uses: AdamKorcz/slsa-github-generator/.github/workflows/builder_gradle_slsa3.yml@make-gradle-publisher-an-action
+    with:
+      rekor-log-public: true
+      artifact-list: build/libs/artifact1-0.1.18.jar,build/libs/artifact-0.1.18-javadoc.jar,build/libs/artifact-0.1.18-sources.jar
+```
+
+To use the Publish action, you need to add another job:
+
+```yaml
+publish:
```

Comment:
The indentation in this yaml seems inconsistent with 4 spaces for some lines and 2 for others.